### DISCUSSION TITLE
How to debug spacehammer in fennel

### DISCUSSION THREAD

**Discussion by zhaogu0qing** [agzam/spacehammer#81] [Q&A] [answered]
i can only use log..

**Comment by jaidetree** [comment, marked as answer]
There are a few options. Have you looked at the fennel-lang reference https://fennel-lang.org/reference or the Hammerspoon API docs https://www.hammerspoon.org/docs/index.html?

The fennel-lang doc gives us `(print)` and the Hammerspoon API docs provides `hs.log` (as you may already know) and `hs.inspect`. 

I use print and hs.inspect myself like

```fennel
(print (hs.inspect some-table))
```

Given that fennel is hosted in lua, some of the lua debug tools may work as well:
https://www.tutorialspoint.com/lua/lua_debugging.htm and confirmed https://groups.google.com/g/hammerspoon/c/x_arf-yI2SU/m/hnnOKyyBDwAJ.

Additionally there is at least one interactive debugger that has been confirmed to work by the Hammerspoon maintainer:
https://github.com/Hammerspoon/hammerspoon/issues/1268

**Reply by zhaogu0qing** [reply]
thanks a lot!

**Comment by Grazfather** [comment]
You can now use the repl. See #128
